TrenchBroom-Benchmark Linker error with clang-6.0 on Ubuntu 17.10

```
/usr/bin/ld: CMakeFiles/common.dir/common/src/Renderer/MapRenderer.cpp.o: undefined reference to symbol 'glDepthFunc'
//usr/lib/x86_64-linux-gnu/mesa/libGL.so.1: error adding symbols: DSO missing from command line
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```
The problem seems to be libGL is not linked to TrenchBroom-Test/TrenchBroom-Benchmark. Copying the bit of cmake that links to libGL from the TrenchBroom app fixes it.

--- a/cmake/TrenchBroomTest.cmake
+++ b/cmake/TrenchBroomTest.cmake
@@ -44,6 +44,14 @@ IF (COMPILER_IS_MSVC)
 	SET_TARGET_PROPERTIES(TrenchBroom-Benchmark PROPERTIES LINK_FLAGS_RELEASE "/DEBUG /PDBSTRIPPED:Release/TrenchBroom-Benchmark-stripped.pdb /PDBALTPATH:TrenchBroom-Benchmark-stripped.pdb")
 ENDIF()
 
+# Properly link to OpenGL libraries on Unix-like systems
+IF(${CMAKE_SYSTEM_NAME} MATCHES "Linux|FreeBSD")
+    FIND_PACKAGE(OpenGL)
+    INCLUDE_DIRECTORIES(SYSTEM ${OPENGL_INCLUDE_DIR})
+    TARGET_LINK_LIBRARIES(TrenchBroom-Test ${OPENGL_LIBRARIES})
+    TARGET_LINK_LIBRARIES(TrenchBroom-Benchmark ${OPENGL_LIBRARIES})
+ENDIF()
+
 SET(RESOURCE_DEST_DIR "$<TARGET_FILE_DIR:TrenchBroom-Test>")
 SET(BENCHMARK_RESOURCE_DEST_DIR "$<TARGET_FILE_DIR:TrenchBroom-Benchmark>")
 

